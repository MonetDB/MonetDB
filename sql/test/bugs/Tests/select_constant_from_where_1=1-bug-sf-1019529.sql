create table t1019529 (name varchar(1024));
insert into t1019529 values ('niels'), ('fabian'), ('martin');

select name from t1019529 where name='doesnotexist';
select 1 from t1019529 where name='doesnotexist';
select 1 from t1019529;
select 1 from t1019529 where 0=1;
select 1 from t1019529 where 1=1;
select 1 where 0=1;
select 1 where 1=1;
select 1, count(*) from t1019529;
select name, count(*) from t1019529; -- should fail
select 1, count(*) from t1019529 where 0=1;
select 1, count(*) where 0=1;

drop table t1019529;
