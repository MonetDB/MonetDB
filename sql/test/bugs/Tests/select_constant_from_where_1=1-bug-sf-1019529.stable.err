stderr of test 'select_constant_from_where_1=1-bug-sf-1019529` in directory 'sql/test/bugs` itself:


# 13:45:14 >  
# 13:45:14 >  Mtimeout -timeout 180 Mserver "--config=/ufs/niels/data/Linux-i686/etc/MonetDB.conf" --debug=10 --set "monet_mod_path=/ufs/niels/data/Linux-i686/lib/MonetDB" --set "gdk_dbfarm=/ufs/niels/data/Linux-i686/var/MonetDB/dbfarm" --set "sql_logdir=/ufs/niels/data/Linux-i686/var/MonetDB/log" --set mapi_port=47399 --set sql_port=57322 --set monet_prompt= --trace "--dbname=mTests_src_test_bugs" "/net/stem.ins.cwi.nl/export/scratch1/niels/Linux-i686/build-sql/src/backends/monet4/sql_server.mil" ; echo ; echo Over..
# 13:45:14 >  


# 13:45:16 >  
# 13:45:16 >  Mtimeout -timeout 60 MapiClient --language=sql -u monetdb -P monetdb --host=stem --port=57322  < select_constant_from_where_1=1-bug-sf-1019529.sql
# 13:45:16 >  

MAPI  = (monetdb) /var/tmp/mtest-27483/.s.monetdb.35395
QUERY = select name, count(*) from t1019529; -- should fail
ERROR = !SELECT: cannot use non GROUP BY column 'name' in query results without an aggregate function
CODE  = 42000

# 13:45:16 >  
# 13:45:16 >  Done.
# 13:45:16 >  

