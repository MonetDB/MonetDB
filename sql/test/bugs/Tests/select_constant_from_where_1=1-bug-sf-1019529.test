statement ok
create table t1019529 (name varchar(1024))

statement ok
insert into t1019529 values ('niels'), ('fabian'), ('martin')

query T rowsort
select name from t1019529 where name='doesnotexist'
----

query I rowsort
select 1 from t1019529 where name='doesnotexist'
----

query I rowsort
select 1 from t1019529
----
1
1
1

query I rowsort
select 1 from t1019529 where 0=1
----

query I rowsort
select 1 from t1019529 where 1=1
----
1
1
1

query I rowsort
select 1 where 0=1
----

query I rowsort
select 1 where 1=1
----
1

query II rowsort
select 1, count(*) from t1019529
----
1
3

statement error
select name, count(*) from t1019529

query II rowsort
select 1, count(*) from t1019529 where 0=1
----
1
0

query II rowsort
select 1, count(*) where 0=1
----
1
0

statement ok
drop table t1019529

