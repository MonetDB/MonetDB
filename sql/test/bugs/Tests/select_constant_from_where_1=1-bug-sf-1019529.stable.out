stdout of test 'select_constant_from_where_1=1-bug-sf-1019529` in directory 'sql/test/bugs` itself:


# 11:58:07 >  
# 11:58:07 >   mserver5 "--config=/ufs/goncalve/scratch/MonetDB/stable/debug/MonetDB5/etc/monetdb5.conf" --debug=10 --set "monet_mod_path=/ufs/goncalve/scratch/MonetDB/stable/debug/MonetDB5//lib/MonetDB5:/ufs/goncalve/scratch/MonetDB/stable/debug/MonetDB5//lib/MonetDB5/lib:/ufs/goncalve/scratch/MonetDB/stable/debug/MonetDB5//lib/MonetDB5/bin" --set "gdk_dbfarm=/ufs/goncalve/scratch/MonetDB/stable/debug/MonetDB5//var/MonetDB5/dbfarm" --set "sql_logdir=/ufs/goncalve/scratch/MonetDB/stable/debug/MonetDB5//var/MonetDB5/sql_logs"  --set mapi_open=true --set xrpc_open=true --set mapi_port=30364 --set xrpc_port=49769 --set monet_prompt= --trace  "--dbname=mTests_src_test_bugs" --set mal_listing=0 "--dbinit= include sql;" ; echo ; echo Over..
# 11:58:07 >  

# MonetDB server v5.6.1, based on kernel v1.24.1
# Serving database 'mTests_src_test_bugs'
# Compiled for x86_64-unknown-linux-gnu/64bit with 64bit OIDs dynamically linked
# Copyright (c) 1993-2008 CWI, all rights reserved
# Visit http://monetdb.cwi.nl/ for further information
# Listening for connection requests on mapi:monetdb://alviss.ins.cwi.nl:30364/
# MonetDB/SQL module v2.24.1 loaded

#function user.main():void;
#    clients.quit();
#end main;


# 12:53:19 >  
# 12:53:19 >  Mtimeout -timeout 60 MapiClient -lsql -u monetdb -P monetdb --host=localhost --port=45579  < select_constant_from_where_1=1-bug-sf-1019529.sql
# 12:53:19 >  

[ 3	]
% sys.t1019529 # table_name
% name # name
% varchar # type
% 0 # length
#select 1 from t1019529 where name='doesnotexist';
% .%1 # table_name
% %1 # name
% tinyint # type
% 1 # length
#select 1 from t1019529;
% .%1 # table_name
% %1 # name
% tinyint # type
% 1 # length
[ 1	]
[ 1	]
[ 1	]
#select 1 from t1019529 where 0=1;
% .%1 # table_name
% %1 # name
% tinyint # type
% 1 # length
#select 1 from t1019529 where 1=1;
% .%1 # table_name
% %1 # name
% tinyint # type
% 1 # length
[ 1	]
[ 1	]
[ 1	]
#select 1 where 0=1;
% .%2 # table_name
% %2 # name
% tinyint # type
% 1 # length
#select 1 where 1=1;
% .%2 # table_name
% %2 # name
% tinyint # type
% 1 # length
[ 1	]
#select 1, count(*) from t1019529;
% .%2,	.%1 # table_name
% %2,	%1 # name
% tinyint,	bigint # type
% 1,	1 # length
[ 1,	3	]
#select 1, count(*) from t1019529 where 0=1;
% .%2,	.%1 # table_name
% %2,	%1 # name
% tinyint,	bigint # type
% 1,	1 # length
[ 1,	0	]
#select 1, count(*) where 0=1;
% .%3,	.%1 # table_name
% %3,	%1 # name
% tinyint,	bigint # type
% 1,	1 # length
[ 1,	0	]

# 12:53:19 >  
# 12:53:19 >  Done.
# 12:53:19 >  

